src/apps/wall/util.lua

```diff
@@ -0,0 +1,89 @@
+local link = require("core.link")
+local ffi  = require("ffi")
+local bit  = require("bit")
+
+local l_receive, l_transmit = link.receive, link.transmit
+local l_nreadable, l_nwritable = link.nreadable, link.nwritable
+
+local uint16_ptr_t = ffi.typeof("uint16_t*")
+local function rd16(address)
+   return ffi.cast(uint16_ptr_t, address)[0]
+end
+
+local uint32_ptr_t = ffi.typeof("uint32_t*")
+local function rd32(address)
+   return ffi.cast(uint16_ptr_t, address)[0]
+end
+
+local function ipv6_addr_cmp(a, b)
+   for i = 1, 16 do
```

Comment:
I'm not sure this function is working as it's intended. `a` and `b` are pointers to an IPv6 address (pointers to a cdata buffer). Shouldn't be `local d = b[i] - a[i]`? At this moment the function is substracting `a` and `b` 16 times (which are pointers to an address). In addition to that, since `a` are `b` cdata pointers, `i` should start in 0 and not 1 I think.

Maybe it's worth to add an unit test for this function. WDYT?